daemon/src/maker_inc_connections.rs

```diff
@@ -106,39 +129,32 @@ impl Actor {
             .map_ok(move |msg| FromTaker { taker_id, msg })
             .map(forward_only_ok::Message);
 
-        let (out_msg_actor_address, mut out_msg_actor_context) = xtra::Context::new(None);
-
-        let (forward_to_cfd, forward_to_cfd_fut) =
+        let (forward_only_ok, forward_only_ok_fut) =
             forward_only_ok::Actor::new(self.taker_msg_channel.clone_channel())
                 .create(None)
                 .run();
-        self.tasks.add(forward_to_cfd_fut);
-
-        // only allow outgoing messages while we are successfully reading incoming ones
-        let heartbeat_interval = self.heartbeat_interval;
-        let taker_disconnected_channel = self.taker_disconnected_channel.clone_channel();
-        self.tasks.add(async move {
-            let mut actor = send_to_socket::Actor::new(write, transport_state.clone());
-
-            let _heartbeat_handle = out_msg_actor_context
-                .notify_interval(heartbeat_interval, || wire::MakerToTaker::Heartbeat)
-                .expect("actor not to shutdown")
-                .spawn_with_handle();
+        taker_tasks.add(forward_only_ok_fut);
+        taker_tasks.add(forward_only_ok.clone().attach_stream(read));
```

Comment:
I would probably not use `attach_stream` and implement a loop myself. You'll need it once you want to dispatch to different addresses based on the message type.

Reply:
> You'll need it once you want to dispatch to different addresses based on the message type.

Is that so? I think we're sending the filtered messages back to this actor, who can now handle them and dispatch to different addresses.

But I agree that there must be a more direct way of doing this that doesn't involve this `attach_stream` and the `forward_only_ok::Actor` in general.

Reply:
Maybe I am not seeing the big picture, I'll checkout the code and have a look!

Reply:
Yeah I think there is quite some unnecessary indirection here (probably introduced by me :joy: ).

I reckon take a step back and see if you can implement all of this functionality in a simpler way. In a nutshell what we need is:

- Forward messages from the `read` side to the `taker_msg_channel`
- Drop the connections on error on either the read or writing side
- Send a heart-beat every X seconds
- Make sure to stop the heartbeat on errors